storage/bucket.go

```diff
@@ -210,12 +211,20 @@ func (b *BucketHandle) Update(ctx context.Context, uattrs BucketAttrsToUpdate) (
 	if uattrs.PredefinedDefaultObjectACL != "" {
 		req.PredefinedDefaultObjectAcl(uattrs.PredefinedDefaultObjectACL)
 	}
-	// TODO(jba): retry iff metagen is set?
-	rb, err := req.Context(ctx).Do()
-	if err != nil {
+
+	isIdempotent := b.conds != nil && b.conds.MetagenerationMatch != 0
```

Comment:
This looks good good for BucketHandle.Update. Could you add the logic to use the retry config to the other direct methods on `BucketHandle` as well? It will be confusing to have it work only on some methods in the mean time.

Reply:
Added!